Added parsing in intervals

dpgrote opened the pull request:
This allows expressions to be used when specifying intervals in the input files.

Since I added `parseStringtoInt`, I also added `parseStringtoReal` and used it to replace duplicated code in the query and get with parser routines for a minor cleanup.

ax3l commented on Source/Utils/WarpXUtil.H:
```diff
@@ -182,6 +182,15 @@ safeCastToInt(amrex::Real x, const std::string& real_name);
+ * \param name For integers, the name, to be used in error messages
+ */
+amrex::Real parseStringtoReal(std::string str);
+int parseStringtoInt(std::string str, std::string name);
```
Technically to parse this well, we unfortunately need to do thisL
```suggestion
/** Parse a string and return as a real
 *
 * In case the string cannot be interpreted as Real,
 * this function ... <throws an exception? aborts with error message?>
 *
 * \param str The string to be parsed
 * \return representation as real
 */
amrex::Real parseStringtoReal(std::string str);

/** Parse a string and return an int
 *
 * In case the string cannot be interpreted as Real,
 * this function ... <throws an exception? aborts with error message?>
  *
 * \param str The string to be parsed
 * \param name For integers, the name, to be used in error messages
 * \return rounded closest integer
 */
int parseStringtoInt(std::string str, std::string name);
```

ax3l replied:
Nit pick: I would probably spell `parseStringtoInt` as `parseStringToInt` and
`parseStringtoReal` as `parseStringToReal`

NeilZaim reviewed (commented):
Thanks a lot for this PR, it's useful and looks good to me. Few comments/questions:

- I don't see why it wouldn't work, but did you test with a syntax that includes the colons? (e.g. `5:last_step:T_period/dt`). We could potentially add something like that in CI as well.
- Could be worth mentioning somewhere in the parameters doc that the parser works with intervals syntax as well (just a couple of words should be enough).

dpgrote commented:
Thanks for your comments!
I added documentation (I had already done this locally but neglected to do the push).
I updated `inputs_2d_multi_rz_rt` to test the expressions with colons.

ax3l reviewed (approved):
:tada:
